src/hotspot/share/jvmci/jvmciRuntime.cpp

```diff
@@ -2175,6 +2175,7 @@ JVMCI::CodeInstallResult JVMCIRuntime::register_method(JVMCIEnv* JVMCIENV,
                                                         nmethod_entry_patch_offset,
                                                         nmethod_mirror_name,
                                                         failed_speculations);
+      nmethod::ResultStatus result_status;
```

Comment:
Please expose the new `out_of_memory` result to JVMCI (e.g. in `JVMCI::CodeInstallResult` enum and `HotSpotVMConfig.getCodeInstallResultDescription` method).